CleverRaven/Cataclysm-DDA
# Unequip armor transformed into non-armor

#### Summary
Bugfixes "Unequip armor transformed into non-armor"


#### Purpose of change
When an armor item transforms into non-armor, the player is then wearing a non-armor item. This causes unpredictable behavior when the game tries to access the item's undefined armor values.  This change addresses this issue.
Fixes #56833


#### Describe the solution
This change modifies the item transformation code to unequip an item when it becomes non-armor.


#### Describe alternatives you've considered
1. Not allowing an equipped item to transform if the transform target is not armor. This would lead to ever-burning joints/cigs.
2. Adding armor values to joint roaches. This would fix the issue, but would need to be done for each transforming item (existing and future) 


#### Testing
1. Start a new punk rock dude
3. Craft joints
4. Light joint
5. Wear joint
6. Extinguish joint
7. Check that the joint roach is properly unequipped


#### Additional context
